nixos/tests/postgresql.nix

```diff
@@ -1,26 +1,44 @@
-import ./make-test.nix ({ pkgs, ...} : {
+import ./make-test.nix ({ pkgs, ...} : 
+let
+  postgresql-versions = pkgs.callPackages ../../pkgs/servers/sql/postgresql { };
+  test-sql = pkgs.writeText "postgresql-test" ''
+    CREATE EXTENSION pgcrypto; -- just to check if lib loading works
+    CREATE TABLE sth (
+      id int
+    );
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+  '';
+in
+{
   name = "postgresql";
   meta = with pkgs.stdenv.lib.maintainers; {
     maintainers = [ zagy ];
   };
 
-  nodes = {
-    master =
-      { pkgs, config, ... }:
-
+  nodes = pkgs.lib.mapAttrs' (k: v: {name=k; value=
+      {pkgs, config, ...}:
       {
+        services.postgresql.package=v;
         services.postgresql.enable = true;
-        services.postgresql.initialScript =  pkgs.writeText "postgresql-init.sql"
-          ''
-          CREATE ROLE postgres WITH superuser login createdb;
-          '';
-      };
-  };
+      };}) 
+      postgresql-versions;
 
-  testScript = ''
-    startAll;
-    $master->waitForUnit("postgresql");
-    $master->sleep(10); # Hopefully this is long enough!!
-    $master->succeed("echo 'select 1' | sudo -u postgres psql");
-  '';
+  testScript = builtins.concatStringsSep "\n" (map (p: ''
+    ${"$"}${p}->start;
```

Comment:
@spinus I'm wondering: since the nodes in this test don't interact with eachother it would be nicer if the individual tests would be executed in parallel. 

There already seems support for this. See the [callSubTests](https://github.com/NixOS/nixpkgs/blob/master/nixos/release.nix#L22) function and how it's used to call [nixos/tests/boot.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/boot.nix) for example.

If you go with this more efficient approach I predict that your `testScript` will become simpler again because there won't be any need for the `${"$"}${p}->...` trickery.